### ISSUE TITLE
Settings panel: add a keyword argument in apply button handler

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#8315]
Hi,

Related to #8310:

There are times when certain settings panels may wish to perform some operations when Apply button is pressed from Settings screen. For example, when UI language has changed and Apply button is pressed, NVDA should not restart.

Thus, a new keyword argument named "applyOnly" is hereby introduced to let panels take note of the difference between OK and Apply buttons. The immediate beneficiary is General category, in that when language has changed and Apply button is pressed, a warning should be shown and NVDA will not restart.

Thanks.

**Comment by leonardder** [comment]
I tend to disagree on the behaviour of apply for the general category you propose. It makes sense to ask the user whether they want to restart NVDA for the changes to take effect, even if they press apply. In any case, any changes or addition of a new handler in this area should be left out of scope for NVDA 2018.2

**Comment by josephsl** [comment]
Hi, but what if, as noted in #8310, users are not given a chance to reassign keyboard modifiers? Thanks.

**Comment by leonardder** [comment]
I wouldn't do preSave, onSave and postSave sequentially for every panel, 
but first loop through all the panels and do preSave, than run all the 
onSave handlers, than all the postSave handlers. This keeps the user 
experience as tight to as it was before, i.e. if pre evaluation fails, 
nothing is saved. In the case of unchecking all modifier keys from the 
keyboard panel, pressing ok would just pop up the error message that 
tells one that at least one modifier has to be set. When dismissed, the 
settings dialog will return without saving anything. In the case of the 
general settings postSave, it will make sure that all settings are saved 
properly before restarting.

**Comment by josephsl** [comment]
Hi, yep, that could work. Thanks.

**Comment by Adriani90** [comment]
cc: @SaschaCowley
